src/signal.rs

```diff
@@ -139,42 +137,49 @@ impl SignalHandler {
 
 /// Signals from OS/user.
 #[cfg(unix)]
-fn os_signals() -> impl Stream<Item = SignalTo> {
+fn os_signals(runtime: &Runtime) -> impl Stream<Item = SignalTo> {
     use tokio::signal::unix::{signal, SignalKind};
 
-    let mut sigint = signal(SignalKind::interrupt()).expect("Signal handlers should not panic.");
-    let mut sigterm = signal(SignalKind::terminate()).expect("Signal handlers should not panic.");
-    let mut sigquit = signal(SignalKind::quit()).expect("Signal handlers should not panic.");
-    let mut sighup = signal(SignalKind::hangup()).expect("Signal handlers should not panic.");
-
-    async_stream::stream! {
-        loop {
-            let signal = tokio::select! {
-                _ = sigint.recv() => {
-                    info!(message = "Signal received.", signal = "SIGINT");
-                    SignalTo::Shutdown
-                },
-                _ = sigterm.recv() => {
-                    info!(message = "Signal received.", signal = "SIGTERM");
-                    SignalTo::Shutdown
-                } ,
-                _ = sigquit.recv() => {
-                    info!(message = "Signal received.", signal = "SIGQUIT");
-                    SignalTo::Quit
-                },
-                _ = sighup.recv() => {
-                    info!(message = "Signal received.", signal = "SIGHUP");
-                    SignalTo::ReloadFromDisk
-                },
-            };
-            yield signal;
+    // The `signal` function must be run within the context of a Tokio runtime.
+    runtime.block_on(async {
+        let mut sigint =
+            signal(SignalKind::interrupt()).expect("Signal handlers should not panic.");
+        let mut sigterm =
+            signal(SignalKind::terminate()).expect("Signal handlers should not panic.");
+        let mut sigquit = signal(SignalKind::quit()).expect("Signal handlers should not panic.");
+        let mut sighup = signal(SignalKind::hangup()).expect("Signal handlers should not panic.");
```

Comment:
Should each of these have unique `expects()` for each `SignalKind`? The panic should let us get back to the right line, but thought it may be useful.

Though I also don't know when just one of these may panic.

Reply:
That was copied, but I can definitely change it up.

Reply:
Sure, wasn't sure if it'd be super useful

Reply:
Done in fab0f1825a1c92b6ed441b9fb046415214f20375